clouddriver-kubernetes-v2/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/manifest/KubernetesRollingRestartManifestOperation.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.kubernetes.v2.op.manifest;
+
+import com.netflix.spinnaker.clouddriver.data.task.Task;
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.KubernetesCoordinates;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.KubernetesResourceProperties;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesRollingRestartManifestDescription;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.op.handler.CanRollingRestart;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.op.handler.KubernetesHandler;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.security.KubernetesV2Credentials;
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation;
+import java.util.List;
+
+public class KubernetesRollingRestartManifestOperation implements AtomicOperation<Void> {
+  private final KubernetesRollingRestartManifestDescription description;
+  private final KubernetesV2Credentials credentials;
+  private static final String OP_NAME = "ROLLING_RESTART_KUBERNETES_MANIFEST";
+
+  public KubernetesRollingRestartManifestOperation(
+      KubernetesRollingRestartManifestDescription description) {
+    this.description = description;
+    this.credentials = (KubernetesV2Credentials) description.getCredentials().getCredentials();
```

Comment:
Looks like all of the V2 operations need to do this casting, but I think we can avoid it by adding a generic `<C extends KubernetesCredentials>` to `KubernetesAtomicOperationDescription` and then making `KubernetesAtomicOperationDescription` be `KubernetesAtomicOperationDescription<KubernetesV2Credentials>`.

Reply:
Thank you for being the type of code reviewer that teaches me a new trick every time! Just updated in a separate commit; will open a separate PR to remove the now-unnecessary casts from the other Kubernetes V2 classes, and add the appropriate param to the V1 only classes.